src/pages/settings/Profile/DisplayNamePage.js

```diff
@@ -57,60 +55,25 @@ class DisplayNamePage extends Component {
      */
     validate(values) {
         const errors = {};
-
-        // Check for invalid characters in first and last name
-        const [firstNameInvalidCharacter, lastNameInvalidCharacter] = ValidationUtils.findInvalidSymbols(
+        const [doesFirstNameHaveInvalidCharacters, doesLastNameHaveInvalidCharacters] = ValidationUtils.doesContainCommaOrSemicolon(
             [values.firstName, values.lastName],
         );
-        this.assignError(
-            errors,
-            'firstName',
-            !_.isEmpty(firstNameInvalidCharacter),
-            Localize.translateLocal(
-                'personalDetails.error.hasInvalidCharacter',
-                {invalidCharacter: firstNameInvalidCharacter},
-            ),
-        );
-        this.assignError(
-            errors,
-            'lastName',
-            !_.isEmpty(lastNameInvalidCharacter),
-            Localize.translateLocal(
-                'personalDetails.error.hasInvalidCharacter',
-                {invalidCharacter: lastNameInvalidCharacter},
-            ),
-        );
-        if (!_.isEmpty(errors)) {
-            return errors;
+
+        // First we validate the first name field
+        if (doesFirstNameHaveInvalidCharacters) {
+            errors.firstName = this.props.translate('personalDetails.error.hasInvalidCharacter');
+        } else if (ValidationUtils.doesContainReservedWord(values.firstName, CONST.DISPLAY_NAME.RESERVED_FIRST_NAMES)) {
+            errors.firstName = this.props.translate('personalDetails.error.containsReservedWord');
```

Comment:
**:issue** We should display multiple error messages at the same time if multiple validations fail, as we state in the [form guidelines](https://github.com/Expensify/App/blob/7e6a776e2e78ba361ab1d944f0835d0c208eea53/contributingGuides/FORMS.md#L134). For example, if I set my first name to `Concierge;` I should see `First name cannot contain the words Expensify or Concierge. Name cannot contain a comma or semicolon.`.

**suggestion:** We can easily join the error messages together.

Reply:
> suggestion: We can easily join the error messages together.

I'm not sure we do this in any of our other forms?

Reply:
Agreed, could you give an example of a form that does this? Because that would require copy variations and conditions for each combination of errors. Or I guess appending the error to `errors.firstName` instead of replacing it.

Reply:
I don't think we currently do this in any of our forms. I'll ask in Slack if we want to do this. If not then I'm happy with what we have and we should update the form guidelines.

Reply:
Here's the [Slack discussion](https://expensify.slack.com/archives/C01GTK53T8Q/p1676403131894039).

Reply:
I'll make this a NAB comment. If we decide to make the change I'll create a refactor issue.

Reply:
Alright sounds good. I'm going to go ahead and merge then. Thanks for starting the slack discussion @neil-marcellini.